[SPARK-7504] [YARN] NullPointerException when initializing SparkContext…

zzvara opened the pull request:
… in YARN-cluster mode

Added check if ApplicationMaster has been initialized with additional error message.

zzvara committed:
SPARK-7504 [YARN] NullPointerException when initializing SparkContext in YARN-cluster mode

Added check if ApplicationMaster has been initialized with additional error message.

AmplabJenkins commented:
Can one of the admins verify this patch?

andrewor14 commented:
ok to test

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #32307 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/32307/consoleFull) for   PR 6029 at commit [`7582ea8`](https://github.com/apache/spark/commit/7582ea864314cb3889ee224c22b5e8f88261c5e6).

SparkQA commented:
[Test build #32307 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/32307/consoleFull) for   PR 6029 at commit [`7582ea8`](https://github.com/apache/spark/commit/7582ea864314cb3889ee224c22b5e8f88261c5e6).
 * This patch **fails Scala style tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/32307/
Test FAILed.

zzvara committed:
SPARK-7504 [YARN] NullPointerException when initializing SparkContext in YARN-cluster mode

Added check if ApplicationMaster has been initialized with additional error message.

zzvara committed:
SPARK-7504 [YARN] NullPointerException when initializing SparkContext in YARN-cluster mode

Added check if ApplicationMaster has been initialized with additional error message.

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #32312 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/32312/consoleFull) for   PR 6029 at commit [`6a6d1b1`](https://github.com/apache/spark/commit/6a6d1b1772cae5fb6a16c19661800d888b514783).

SparkQA commented:
[Test build #32312 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/32312/consoleFull) for   PR 6029 at commit [`6a6d1b1`](https://github.com/apache/spark/commit/6a6d1b1772cae5fb6a16c19661800d888b514783).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/32312/
Test PASSed.

srowen commented on yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala:
```diff
@@ -573,10 +573,19 @@ object ApplicationMaster extends Logging {
   }
 
   private[spark] def sparkContextInitialized(sc: SparkContext): Unit = {
+    if(master == null) {
```
(Should be a space after `if` but that can be fixed on merge; might fix that if you make any other change.) 

I think the text is OK although I suppose that as of 1.4 there's the launcher API, but, probably not worth writing a man page here.

ehnalis replied:
I'll just let you fix it on merge.

Also, thank you for attracting my attention to the launcher API.

andrewor14 commented:
I think @vanzin left some comments on the JIRA that should be addressed here.

zzvara committed:
SPARK-7533 [YARN] Decrease spacing between AM-RM heartbeats.

Added a new configuration parameter to tailor heartbeat intervals upon pending allocations.

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #32499 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/32499/consoleFull) for   PR 6029 at commit [`1f63dbb`](https://github.com/apache/spark/commit/1f63dbb652191cf8f6f6d36a5b5ec6b7f713184f).

ehnalis commented:
@andrewor14 Can I use this PR for two issues?

srowen commented:
Eh, this now seems to have some unrelated changes. These are quite unrelated JIRAs, so you certainly shouldn't make one PR for both. 

I think SPARK-7504 should just be closed, along with this PR, and a new one can be opened for your other separate changes. I am not sure 7504 is worth bothering making a new exception for.

SparkQA commented:
[Test build #32499 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/32499/consoleFull) for   PR 6029 at commit [`1f63dbb`](https://github.com/apache/spark/commit/1f63dbb652191cf8f6f6d36a5b5ec6b7f713184f).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/32499/
Test PASSed.
